### DISCUSSION TITLE
Arduino 2560 - Delay/Lag on release on digital inputs

### DISCUSSION THREAD

**Discussion by nicolasvair** [shanteacontrols/OpenDeck#154] [Q&A] [answered]
Hello ! 

First of all a gigantic thank you for this formidable tool ! 

We are building a music spaceship with two Arduino 2560, and currently have some latency on some buttons, but only on the release of the button. When we push it, the midi message is sent immediately. When we release, it take between 1  and 4 sec to send the note off signal.

It seems like the Arduino have trouble to get back to the 5v value. Even more weird, when we connect a voltmeter to the 2 pins of the button, it get stuck in ON state, and it's only when we remove the voltmeter that it get back to off after 2 sec.

It only happen on some digital inputs.

It gets even more weird because when we use a cable to connect make the connection between the two sides of the button, we don't have this problem..  We've been searching for a while, we thought it might be the buttons first, but we changed it and experience exactly the same issue. 

Would you have any hints on where we should continue searching ? Could it be related to wiring ?

Here are some pictures of this uncommon "controller" 😁

![IMG_0142](https://user-images.githubusercontent.com/79313360/170026878-9205bae6-96bb-4a36-92c9-8f74dbe3d492.JPG)

![IMG_0141](https://user-images.githubusercontent.com/79313360/170027266-08cd3226-402d-4bce-a9c5-2b79436aef70.JPG)

Thank you !

PS : I saw the other post about lag on analog inputs on mega 2560 but it doesn't seemed related.

**Comment by paradajz** [comment]
Hey,

the spaceship looks crazy. :)

Do you know which pins are related? You said you were using two Arduinos, is the issue the same on both of them, and if yes, is it always the same pins?

**Reply by nicolasvair** [reply]
Thank you for the super short reply !

One thing to add, this morning, when we first started using the program we had no problem, it appeared again after 5/10mn of use.

On the first Arduino, we have the problem on midi CC - inputs (we kept the default numbering, changed midi note to CC/0 Off) : 
CC1 - D52
CC17 - D51
CC16 - D53
CC6 - D40
CC22 - D41
CC23 - D39

On the second Arduino, we have the problem on midi CC - inputs
CC1 - D52
CC17 - D51
CC16 - D53
CC6 - D40
CC22 - D41
CC23 - D39

So yes it's exactly the same pins on both units.

We are running the 7.0.0 (if possible it would be much more simple for us not flashing - others Arduino are used in other attractions)

Hvala lepo

**Reply by nicolasvair** [reply]
It look like an electrostatic phenomenon. When I'm touching the cables near the Arduino the time reaction is much shorter..
Any ideas how I should test this / solve this ?

**Reply by paradajz** [reply]
Pretty sure it's a wiring problem then. How does your wiring look like? What's the length of wires? Are the wires passing near some noisy lines? Is your ground okay?

**Reply by paradajz** [reply]
Also, for test purposes, can you try using one Mega board with just a single button connected to one of the problematic pins outside your project?

**Reply by nicolasvair** [reply]
Hello, 

After some hours more of search, it seems to be related to a microphone plugged in the sound card. We don't have the issue when the mic is unplugged.
Don't know yet if it's related to the 48v or some grounding issue.. Will do some test tomorrow and keep you updated if we found the solution, but it seems we are on track 🙏

If you have any ideas feel free to share !

Thank you

**Comment by nicolasvair** [comment]
Quick update

We thought we had found the solution :
We had one mic that was connected to ground, removing it suppressed the issue, so we isolated it from the metallic structure. 

It worked ok but now that we are on another electrical installation we have the same’ issue. Still on the same buttons. 

It’s definitely related to ground but it’s hard to find a solution. 

We are using a Mac mini that isn’t connected to ground (only 2 wire power adapter). When using only the Mac mini and the arduino we have no problem. 

Then if we plug a screen (connected to ground) or sound card (connected to an amp connected to ground), then the issue is back. 

We are not sure how to deal with this so still searching.. 

We will post if we finally find a solution (I’m sure we will !)

**Comment by nicolasvair** [comment, marked as answer]
Hi ! So after much more tries, it seems indeed to be related to a fault in grounding. 

The Mac mini that we are using has no ground. So, after some time we see a voltage around 30V between ground of the computer (and so for the Arduino), and the "true" ground (that is used elsewhere in the circuit).

I can't explain how but it was preventing open deck to run smoothly (buttons would stay "on" and not go back to zero, or after some time - so we first identified it as a lag).

We ended up "grounding" each problematic button with a resistor (like on the analog inputs).

It has been running smoothly for a month, varying places and electrical installation, so hopefully we can consider this solved ! 😚💨👻


Here is a quick video of the spaceship running :
https://user-images.githubusercontent.com/79313360/177651264-2ad8dd36-eb2c-4a6e-a3c4-fc83924d8f6f.mov


Hvala lepo ciao
